File: Examples/Ruby/Worksheet/get_worksheet_autoshape_with_format.rb
require 'aspose_cells_cloud'

class Worksheet

  include AsposeCellsCloud
  include AsposeStorageCloud

  def initialize
    #Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("APP_KEY", "APP_SID")
    @cells_api = CellsApi.new  
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new 
    response = @storage_api.put_create(file_name, File.open("../data/" << file_name,"r") { |io| io.read } )
  end

  def get_worksheet_autoshape_with_format
    file_name = "Sample_Test_Book.xls"
    upload_file(file_name)

    sheet_name = "Sheet4"
    autoshape_number = 1
    format = "png"
    response = @cells_api.get_worksheet_autoshape_with_format(file_name, sheet_name, autoshape_number, format)
  end

end

worksheet = Worksheet.new()
puts worksheet.get_worksheet_autoshape_with_format

File: Examples/PHP/Images/GetWorksheetAutoshapeWithFormat.php
//ExStart:
<?php
require_once realpath(__DIR__ . '/..') . '/vendor/autoload.php';
require_once realpath(__DIR__ . '/..') . '/Utils.php';

use Aspose\Cells\CellsApi;
use Aspose\Cells\AsposeApp;

class Image {

	public $cells;

	public function __construct() {
		AsposeApp::$appSID = Utils::appSID;
		AsposeApp::$apiKey = Utils::apiKey;
		$this->cells = new CellsApi();
	}

	public function getWorksheetAutoshapeWithFormat() {
		// Upload file to Aspose Cloud Storage
		$fileName = "Sample_Test_Book.xls";
		Utils::uploadFile($fileName);

        $result = $this->cells->GetWorksheetAutoshapeWithFormat($name=$fileName, $sheetName="Sheet4", $autoshapeNumber=1, $format="png", $storage = null, $folder = null);
        $fh = fopen(realpath(__DIR__ . '/../../..') . '/Data/Output/Autoshape.png', 'w');
        fwrite($fh, $result);
        fclose($fh);
	}
}

$image = new Image();
$image->getWorksheetAutoshapeWithFormat();

?>
//ExEnd:
